gwctl/pkg/policymanager/manager.go

```diff
@@ -320,18 +348,28 @@ func (p Policy) Spec() map[string]interface{} {
 	if err != nil || !ok {
 		return nil
 	}
-
 	result, ok := spec.(map[string]interface{})
 	if !ok {
 		return nil
 	}
+
```

Comment:
A bit out of scope, but we could probably use [`unstructured.NestedMap`](https://pkg.go.dev/k8s.io/apimachinery@v0.30.2/pkg/apis/meta/v1/unstructured#NestedMap) instead of `unstructured.NestedFieldCopy` at line 347 and thus save the extra type casting.